Lib/pdb.py

```diff
@@ -545,15 +547,71 @@ def _disable_tab_completion(self):
         else:
             yield
 
+    def _exec_in_closure(self, source, globals, locals):
+        """ Run source code in closure so code object created within source
+            can find variables in locals correctly
+        """
+
+        # If the source is an expression, we need to print its value
+        try:
+            compile(source, "<string>", "eval")
+            source = "__pdb_eval_result__ = " + source
+            locals["__pdb_eval_result__"] = None
+        except SyntaxError:
+            pass
```

Comment:
```suggestion
        try:
            compile(source, "<string>", "eval")
        except SyntaxError:
            pass
        else:
            source = "__pdb_eval_result__ = " + source
            locals["__pdb_eval_result__"] = None
```

Reply:
Also, why it `__pdb_eval_result__` being set to `None`? You handle the nonexistent case when you pop it later.